Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP.py

```diff
@@ -0,0 +1,3560 @@
+            last_fetch_timestamp = incident_timestamp
+
+    next_run = {
+        'last_fetch': timestamp_to_datestring(last_fetch_timestamp),
+        'previous_ids': list(previous_ids)
+    }
+
+    return next_run, incidents
+
+
+def test_module(client: Client) -> str:
+    """
+    Tests API connectivity and authentication
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    Args:
+        client (Client): Cisco AMP client to run desired requests
+
+    Returns:
+        str: 'ok' if test passed, anything else will fail the test.
+    """
+    try:
+        client.version_get_request()
+
+    except DemistoException as exc:
+        if exc.res and exc.res.status_code == HTTPStatus.UNAUTHORIZED:
+            return 'Authorization Error: Unknown API key or Client ID'
+
+        return exc.message
+
+    return 'ok'
+
+
+def computer_list_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
+    """
+    Get information about computers.
+    The command can get a list of filtered computers or a specific computer with connector_guid.
+    The command supports pagination.
+    If needed the response will be concatenated.
+
+    Args:
+        client (Client): Cisco AMP client to run desired requests
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Raises:
+        ValueError: If there is a connector_guid other arguments must not exist.
+
+    Returns:
+        List[CommandResults]: Information about a list of computers or a specific computer.
+    """
+    page = arg_to_number(args.get('page', 0))
+    page_size = arg_to_number(args.get('page_size', 0))
+    limit = arg_to_number(args.get('limit', 0))
+    connector_guid = args.get('connector_guid', '')
+    hostnames = argToList(args.get('hostname'))
+    internal_ip = args.get('internal_ip')
+    external_ip = args.get('external_ip')
+    group_guids = argToList(args.get('group_guid'))
+    last_seen_within = arg_to_number(args.get('last_seen_within'))
+    last_seen_over = arg_to_number(args.get('last_seen_over'))
+
+    is_get_request = bool(connector_guid)
+    is_list_request = any((
+        page,
+        page_size,
+        limit,
+        hostnames,
+        internal_ip,
+        external_ip,
+        group_guids,
+        last_seen_within,
+        last_seen_over,
+    ))
+
+    if is_get_request and is_list_request:
+        raise ValueError('connector_guid must be the only input, when fetching a specific computer.')
+
+    if not is_get_request:
+        pagination = get_pagination_parameters(page, page_size, limit)
+        raw_response_list: List[Dict[str, Any]] = []
+
+        # Run multiple requests according to pagination inputs.
+        for request_number in \
+                range(pagination.offset_multiplier, pagination.number_of_requests + pagination.offset_multiplier):
```

Comment:
this row is repeated 11 times, please give replace `pagination` with a dataclass, implement an `__iter__` method that returns this range, then `for request number in pagination` would be readable and maintainable (if there's a bug, we can fix it in one place, not 11)


... or just create a method called `pagination_range(pagination: Pagination)`, which will have a very similar maintainability and less work converting

Reply:
Thanks, I've chosen the 2nd suggestion and added the requested method.